rosidl_runtime_py/rosidl_runtime_py/set_message.py

```diff
@@ -33,4 +39,13 @@ def set_message_fields(msg: Any, values: Dict[str, str]) -> None:
         except TypeError:
             value = field_type()
             set_message_fields(value, field_value)
+        rosidl_type = get_message_slot_types(msg)[field_name]
+        # Check if field is an array of ROS messages
+        if isinstance(rosidl_type, Array) or isinstance(rosidl_type, Sequence):
```

Comment:
You can check for `isinstance(rosidl_type, NestedType)` which is the parent class of these two.

Reply:
`NestedType` is the one I needed :+1:  thanks!